### DISCUSSION TITLE
Working example of airplay & google play working in react

### DISCUSSION THREAD

**Discussion by dabobert** [vidstack/player#1208] [Q&A] [answered]
I know that [vidstack supports airplay and googlecast](https://twitter.com/mihar_22/status/1744167648819355967?s=20), and that's one of the main reasons why I implemented it for my project.  I'm an experienced backend dev, but I'm new to react and I'm having a heck of time getting airplay working.  I was hoping someone could post an example so i can understand what i'm doing wrong. I'm using the default layout, but I don't care which template is used I'm just baffled as to what I am doing wrong

**Comment by dabobert** [comment]
If @mihar-22  could just post the sample code of from his video that would be great

**Comment by mihar-22** [comment]
There's nothing to add for AirPlay and Google Cast support in the Default Layout. They should show up if it's on a supported device and there's a remote playback target available.

**Reply by mihar-22** [reply]
I'm testing it right now on both my Mac and iPhone and it works. It doesn't show up in the simulator for some reason, it might require messing with settings.

**Reply by dabobert** [reply]
Two questions
1.  Do you know what setitngs those might be?
2. I believe the default video player may support google cast and airplay, but do you know what changes must be made to the default audio player for it to support both of thse

**Reply by mihar-22** [reply, marked as answer]
1. No idea, not sure if AirPlay works in the simulator at all. It might be a Safari setting to allow access, need to dig around.

2. I just added AirPlay support for audio which will be available in the next release. We don't support Google Cast for audio yet, not sure if it's possible.

**Comment by dabobert** [comment]
1. Ok, i can confirm/replicate that airplay only works in safari and googlecast only seems to work for chromium browsers.  This makes TOTAL sense but i assumed i was doing something wrong due to my limited react skillset
2. This is good to know, I again assumed i was doing something wrong.

Keep up the AMAZING works!
